src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientImpl.java

```diff
@@ -290,10 +290,16 @@ private void configureHttpClient() throws Exception {
         // We use a pooled connection manager to support concurrent threads and connections
         RegistryBuilder<ConnectionSocketFactory> registryBuilder = RegistryBuilder.<ConnectionSocketFactory>create()
             .register("https", sslsf);
-        if (configuration.allowHttpProtocol()) {
-            LOGGER.warn("Insecure HTTP communication is allowed. This should NOT be done on production systems!");
+
+        boolean useProxy = StringUtils.isNotEmpty(System.getProperty("http.proxyHost"));
```

Comment:
This will only work for environments where the proxy is set by the system, not for setups where the proxy configuration is set (only) using OSGI configurations. It also does not account for cases where the configured host is excluded by the proxy. 

Wouldn't it make sense to deprecate and ignore the allowHttpProtocol configuration completely and add the http socket factory always? The warning for insecure connections could still be logged when the configured endpoint is using http protocol.